### ISSUE TITLE
Should Flask-Admin use Flask-WTF?

### ISSUE THREAD

**Issue by longhotsummer** [flask-admin/flask-admin#829]
I'd like to suggest that flask-admin uses flask-wtf by default for its forms. Primarily, this would mean that flask-admin gets CSRF support out the box and is secure by default, which is a good thing.

Flask-WTF does all the hard work to make CSRF work, flask-admin (or users) shouldn't have to duplicate that work.

I know there were some issues blocking adoption of WTForms 2 but those appear to be resolved. I'm using `wtforms==2.0.2`, `flask-admin==1.1.0` and `flask-wtf==0.11` without any hassles. Is there anything technically stopping flask-admin from doing this?

I currently use this code block to ensure that I get CSRF support from flask-wtf:

```python
# Our base form extends flask_wtf.Form to get CSRF support,
# and adds the _obj property required by Flask Admin
class BaseForm(flask_wtf.Form):
    def __init__(self, formdata=None, obj=None, prefix=u'', **kwargs):
        self._obj = obj
        super(BaseForm, self).__init__(formdata=formdata, obj=obj, prefix=prefix, **kwargs)

class MyModelView(ModelView):
    form_base_class = BaseForm
```

**Comment by longhotsummer** [comment]
Btw, I'm happy to help make the required code changes, I wanted to check with the community before doing so.

**Comment by mrjoes** [comment]
Originally flask-admin was built on top of flask-wtf, but there were some significant changes to the flask-wtf API (it stopped exporting fields, etc).

Instead of building on top of flask-wtf, decision was made to make them coexist together. There's property called `form_base_class` http://flask-admin.readthedocs.org/en/v1.0.9/api/mod_model/#flask.ext.admin.model.BaseModelView.form_base_class which can be used to override form class. All templates automatically spit out CSRF fields, if they're defined in the form.

**Comment by longhotsummer** [comment]
I'm interested in making flask-admin secure by default, with the goal that a user should need to do as little as possible to ensure that it runs securely.

The only step that seems to be required is to change the Flask-Admin's BaseForm base class to inherit from Flask-WTF's Form class. The use of wtforms fields can remain unchanged, so they wouldn't be tightly bound at all.

That said, the only reason I had to make the change that I did (described above) was because the inline delete form doesn't get csrf otherwise tokens. I can look into that and see if there is an elegant fix.
